SocialCareCaseViewerApi.Tests/V1/UseCase/SubmissionUseCaseTests.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using FluentAssertions;
+using Moq;
+using NUnit.Framework;
+using SocialCareCaseViewerApi.Tests.V1.Helpers;
+using SocialCareCaseViewerApi.V1.Exceptions;
+using SocialCareCaseViewerApi.V1.Factories;
+using SocialCareCaseViewerApi.V1.Gateways;
+using SocialCareCaseViewerApi.V1.Infrastructure;
+using SocialCareCaseViewerApi.V1.UseCase;
+
+namespace SocialCareCaseViewerApi.Tests.V1.UseCase
+{
+    [TestFixture]
+    public class SubmissionUseCaseTests
+    {
+        private Mock<IDatabaseGateway> _mockDatabaseGateway;
+        private Mock<IMongoGateway> _mockMongoGateway;
+        private SubmissionsUseCase _submissionsUseCase;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _mockDatabaseGateway = new Mock<IDatabaseGateway>();
+            _mockMongoGateway = new Mock<IMongoGateway>();
+            _submissionsUseCase = new SubmissionsUseCase(_mockDatabaseGateway.Object, _mockMongoGateway.Object);
+        }
+
+        [Test]
+        public void ExecutePostSuccessfully()
+        {
+            var request = TestHelpers.CreateCaseSubmissionRequest();
+            var worker = TestHelpers.CreateWorker();
+            var resident = TestHelpers.CreatePerson();
+
+            _mockDatabaseGateway.Setup(x => x.GetWorkerByEmail(request.CreatedBy)).Returns(worker);
+            _mockDatabaseGateway.Setup(x => x.GetPersonByMosaicId(request.ResidentId)).Returns(resident);
+            _mockMongoGateway.Setup(x => x.InsertRecord(It.IsAny<string>(), It.IsAny<CaseSubmission>())).Verifiable();
+
+            var response = _submissionsUseCase.ExecutePost(request);
+            var expectedResponse = TestHelpers.CreateCaseSubmission(SubmissionState.InProgress, response.CreatedAt, worker, resident, response.FormId).ToDomain().ToResponse();
+
+            // struggle to verify _mockMongoGateway.InsertRecord as we do not have access to the dynamically created CaseSubmission object inserted
+            _mockDatabaseGateway.Verify(x => x.GetWorkerByEmail(request.CreatedBy), Times.Once);
+            _mockDatabaseGateway.Verify(x => x.GetPersonByMosaicId(request.ResidentId), Times.Once);
+
+            response.Should().BeEquivalentTo(expectedResponse);
```

Comment:
As discussed, let's see if we can spy on the `InsertRecord`.

Reply:
I update the SubmissionsUseCase#ExecutePost function to return both the response object but also the infrastructure CaseSubmission object, this has allowed InsertRecord to be spied on.

I don't love this solution but it works, there's no real negative it just feels a little ugly.